standalone/prouter/src/i2pd.rs

```diff
@@ -0,0 +1,484 @@
+    let ident: String = c_str_ident.to_str()?.to_owned();
+
+    Ok(ident)
+}
+
+fn get_server_tunnel_name_by_id(index: i32) -> Result<String> {
+    let ptr_name: *const c_char = unsafe { C_GetServerTunnelsName(index) };
+    if ptr_name.is_null() {
+        error!("The name returned from C is corrupted");
+        return Err(anyhow!("The name returned from C is corrupted"));
+    }
+
+    let c_str_name = unsafe { CStr::from_ptr(ptr_name) };
+    let name: String = c_str_name.to_str()?.to_owned();
+
+    Ok(name)
+}
+
+fn get_server_tunnel_ident_by_id(index: i32) -> Result<String> {
+    let ptr_ident: *const c_char = unsafe { C_GetServerTunnelsIdent(index) };
+    if ptr_ident.is_null() {
+        error!("The ident returned from C is corrupted");
+        return Err(anyhow!("The ident returned from C is corrupted"));
+    }
+
+    let c_str_ident = unsafe { CStr::from_ptr(ptr_ident) };
+    let ident: String = c_str_ident.to_str()?.to_owned();
+
+    Ok(ident)
+}
+
+impl I2PD {
+    pub fn new(app_name: String) -> I2PD {
+        I2PD {
+            argc: 1,
+            argv: HashMap::new(),
+            app_name,
+            is_running: false,
+        }
+    }
+
+    fn drop(&mut self) {
+        if self.is_running {
+            self.stop();
+        }
+    }
+
+    fn argv_to_string(&self) -> String {
+        let mut concat_string: String = String::from("./fake-argv0"); // Fake argv input
+        for (key, data) in &self.argv {
+            concat_string.push_str(format!(" --{}={}", key, data).as_str());
+        }
+        info!("PRouter call parameter: {:?}", &concat_string);
+
+        concat_string
+    }
+
+    pub fn add_config(&mut self, key: String, data: String) {
+        self.argv.insert(key, data);
+        self.argc += 1;
+    }
+
+    ///
+    /// Basic control API for I2PD
+    ///
+
+    pub fn init(&self) {
+        if self.is_running {
+            return;
+        }
+        let args_str = self.argv_to_string();
+        let args_c_str =
+            CString::new(args_str).expect("String should be able to be converted into CString");
+        let c_args: *mut c_char = args_c_str.as_ptr() as *mut c_char;
+
+        let name_c_str = CString::new(self.app_name.clone())
+            .expect("String should be able to be converted into CString");
+        let c_name: *const c_char = name_c_str.as_ptr() as *const c_char;
+        unsafe {
+            C_InitI2P(self.argc, c_args, c_name);
+        }
+    }
+
+    pub fn start(&mut self) {
+        if self.is_running {
+            return;
+        }
+        self.is_running = true;
+        unsafe { C_StartI2P() };
+    }
+
+    pub fn close_accepts_tunnels(&self) {
+        if !self.is_running {
+            return;
+        }
+        unsafe { C_CloseAcceptsTunnels() };
+    }
+
+    pub fn stop(&mut self) {
+        if !self.is_running {
+            return;
+        }
+        self.is_running = false;
+        unsafe { C_StopI2P() };
+    }
+
+    ///
```

Comment:
The triple-slash means docstring. Here you just use it as the normal comment. So you should use double-slash instead